pyanaconda/modules/payloads/source/harddrive/harddrive.py

```diff
@@ -43,8 +41,8 @@ def __init__(self):
         self._device = ""
         self.device_changed = Signal()
         self._install_tree_path = ""
-        self._device_mount = INSTALL_TREE + "_device"
-        self._iso_mount = INSTALL_TREE + "_iso"
+        self._device_mount = self._mount_point + "-device"
```

Comment:
Maybe I would override the `mount_point` property to return the device mount point. Then you can use the `super().is_ready` method.

Reply:
Not sure if this suggestion is not bending the purpose of the `MountingSourceBase` class too much.

Reply:
Maybe the real problem here is whether harddrive belongs with other mountables or not?

In a very broad sense it does fit: it mounts, and wants a number. Then again it has multiple mount points, actually a variable number at that, so the actual functionality is different.

Reply:
If the `mount_point` property cannot return a real mount point, then `HardDriveSourceModule` shouldn't be a subclass of `MountingSourceBase`. However, we need to be able to generate the "base" mount point. I still think that there should be a separate class called `MountPointGenerator` with one class method `create_mount_point` that will return a unique mount point. There might be more cases like this one in the future.